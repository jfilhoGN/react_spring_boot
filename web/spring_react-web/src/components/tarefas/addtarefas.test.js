import React from 'react';
import {shallow} from 'enzyme';
import AddTarefas from './addtarefas';


describe('AddTarefas', () => {
    
    test('render component', () => {
        const history = {
            goBack: jest.fn(),
        }
        global.sessionStorage.getItem.mockReturnValue('{"id": "28A", "token":"eyJ0eX"}')
        const wrapper = shallow(<AddTarefas history={history} match={{params: {id: 1}}}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('values of state', () => {
        const wrapper = shallow(<AddTarefas history={history} match={{params: {id: 1}}}/>)
        expect(wrapper.state).toMatchSnapshot();
    });

    test('input text value titulo', () => {
        const wrapper = shallow(<AddTarefas match={{params: {id: 1}}}/>);
        wrapper.find('#titulo').simulate('change', { target: { value: 'titulo'} });
        expect(wrapper.state().titulo).toEqual('titulo');
    });

    test('input text value descricao', () => {
        const wrapper = shallow(<AddTarefas match={{params: {id: 1}}}/>);
        wrapper.find('#descricao').simulate('change', { target: { value: 'descricao descricao'} });
        expect(wrapper.state().descricao).toEqual('descricao descricao');
    });

    test('input text value status', () => {
        const wrapper = shallow(<AddTarefas match={{params:{id:1}}}/>);
        wrapper.find('.status').simulate('change', { value: 1} );
        expect(wrapper.state().status).toEqual(1);
    });

    test('button addTarefa', () => {
        //arrumar
    });

    test('input text editTitulo', () => {
        const wrapper = shallow(<AddTarefas match={{params:{id:1}}}/>);
        wrapper.setState({hasEdit:true})
        wrapper.find('#editTitulo').simulate('change',{target:{value: 'Titulo titulo'}});
        expect(wrapper.state().titulo).toEqual('Titulo titulo')
    });

    test('input text editDescricao', () => {
        const wrapper = shallow(<AddTarefas match={{params:{id:1}}}/>);
        wrapper.setState({hasEdit:true})
        wrapper.find('#editDescricao').simulate('change',{target:{value: 'Descricao descricao'}});
        expect(wrapper.state().descricao).toEqual('Descricao descricao');
    });

    test('input text editStatus', () => {
        const wrapper = shallow(<AddTarefas match={{params:{id:1}}}/>);
        wrapper.setState({hasEdit:true})
        wrapper.find('#editTitulo').simulate('change',{target:{value: 'Titulo titulo'}});
        expect(wrapper.state().titulo).toEqual('Titulo titulo')
    });

    test('button editTarefa', () => {
        //arrumar
    });

    test('getFuncionario()', () => {
        //arrumar
    });

    // ver se handleTituloTarefa, descricao e status
    // necessitam fazer teste
    // componentDidMount
    // handleReturnAddTarefa
    // handlereturnTarefa

    test('handleSubmitTarefa()', () => {
        //arrumar
    });

    test('getDados()', () => {
        //arrumar
    });

    test('handleUpdate()', () => {
        //arrumar
    });



});