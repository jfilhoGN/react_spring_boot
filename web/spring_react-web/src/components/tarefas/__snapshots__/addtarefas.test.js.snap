// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTarefas render component 1`] = `
<Card>
  <Row>
    <form
      className="col s12"
      onSubmit={[Function]}
    >
      <Input
        checked={false}
        id="titulo"
        label="Título"
        name="titulo"
        onChange={[Function]}
        placeholder="Arrumar Cadastro"
        s={12}
        type="text"
      >
        <Icon
          small={true}
        >
          title
        </Icon>
      </Input>
      <Input
        checked={false}
        id="descricao"
        label="Descrição"
        name="descricao"
        onChange={[Function]}
        placeholder="criar um campo com cores"
        s={12}
        type="text"
      >
        <Icon
          small={true}
        >
          description
        </Icon>
      </Input>
      <Col
        m={12}
        s={12}
      >
        <StateManager
          className="status"
          defaultInputValue=""
          defaultMenuIsOpen={false}
          defaultValue={null}
          onChange={[Function]}
          options={
            Array [
              Object {
                "label": "Fazer",
                "value": 1,
              },
              Object {
                "label": "Fazendo",
                "value": 2,
              },
              Object {
                "label": "Em Teste",
                "value": 4,
              },
              Object {
                "label": "Feito",
                "value": 5,
              },
            ]
          }
          placeholder="Status da Tarefa"
          s={12}
        />
        <br />
      </Col>
      <Button
        className="btn-espaco btn-small grey darken-3"
        name="action"
        node="button"
        onClick={[Function]}
        type="submit"
      >
        <i
          className="material-icons"
        >
          add
        </i>
      </Button>
    </form>
  </Row>
</Card>
`;

exports[`AddTarefas values of state 1`] = `[Function]`;
